### PR TITLE
Add `theme.button.intelligentPad`

### LINKED ISSUE #7299: Provide theme opt-out of Button padding logic
### Is this something you or your team is interested in contributing a pull request towards?

Yes.

### Expected Behavior

Grommet theme should provide ability to opt-out of the automatic logic that subtracts border width from button padding. This will allow us to define tech-stack agnostic tokens that can apply to grommet and other frameworks.

### Actual Behavior

StyledButtonKind automatically subtracts border width from button padding.

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR adds in a way to opt out of using our `adjustedPad` logic
#### Where should the reviewer start?
styledButtonKind.js
#### What testing has been done on this PR?
locally and added test
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
opening to naming suggestions
#### What are the relevant issues?
closes #7299 
#### Do the grommet docs need to be updated?
yes
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Button/__tests__/Button-kind-test.js [hunk 1]
```diff
@@ -300,6 +300,42 @@ describe('Button kind', () => {
...
+        },
+        color: 'text-strong',
+        font: {
+          weight: 600,
+        },
+      },
+    },
+  };
+
+  test('padding is applied correctly when intelligentPad is true', () => {
```
Comment (jcfilben) [inline src/js/components/Button/__tests__/Button-kind-test.js:329]: Can we also have a test where intelligentPad is false?

File: src/js/components/Button/StyledButtonKind.js [hunk 2]
```diff
@@ -113,11 +113,18 @@ const getPath = (theme, path) => {
...
+    return css`
+      padding: ${Math.max(parseMetricToNum(pad.vertical) - offset, 0)}px
+        ${Math.max(parseMetricToNum(pad.horizontal) - offset, 0)}px;
+    `;
+  }
+  // If intelligentPad is false return padding without adjustment
   return css`
-    padding: ${Math.max(parseMetricToNum(pad.vertical) - offset, 0)}px
-      ${Math.max(parseMetricToNum(pad.horizontal) - offset, 0)}px;
+    padding: ${pad.vertical} ${pad.horizontal};
```
Comment (jcfilben) [inline src/js/components/Button/StyledButtonKind.js:127]: Do we need to add `px`? I saw on the css snippet above we added it there so wasn't sure if we also needed it here
`padding: ${pad.vertical}px ${pad.horizontal}px;`

Comment (taysea) [reply src/js/components/Button/StyledButtonKind.js:127]: I think in the previous snippet we've parsed metric to num, so we're adding 'px' back. That said, Brittany can you confirm what we have documented in terms of can "pad" be a px and/or t-shirt size? If so, just making sure the t-shirt size works.

Comment (britt6612) [reply src/js/components/Button/StyledButtonKind.js:127]: correct 
``` previous snippet we've parsed metric to num, so we're adding 'px' back. ```

however that said this does not support `t-shirt` sizes however this is a current issue

File: src/js/components/Button/__tests__/Button-kind-test.js (hunk 1 above)
Comment (britt6612) [reply src/js/components/Button/__tests__/Button-kind-test.js:328]: added

File: src/js/components/Button/StyledButtonKind.js (hunk 2 above)
Comment (jcfilben) [reply src/js/components/Button/StyledButtonKind.js:127]: Ok good to know, thanks

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by taysea — APPROVED** [review]

**Review by halocline — APPROVED** [review]

**Pull request merged**